src/runtime_src/core/common/api/xrt_bo.cpp

```diff
@@ -319,12 +319,12 @@ alloc(xclDeviceHandle dhdl, size_t sz, xrtBufferFlags flags, xrtMemoryGroup grp)
   switch (type) {
   case 0:
     return alloc_hbuf(dhdl, xrt_core::aligned_alloc(get_alignment(), sz), sz, flags, grp);
-  case XCL_BO_FLAGS_CACHEABLE:
-  case XCL_BO_FLAGS_SVM:
-  case XCL_BO_FLAGS_DEV_ONLY:
-  case XCL_BO_FLAGS_HOST_ONLY:
-  case XCL_BO_FLAGS_P2P:
-  case XCL_BO_FLAGS_EXECBUF:
+  case (xrtBufferFlags)XCL_BO_FLAGS_CACHEABLE:
+  case (xrtBufferFlags)XCL_BO_FLAGS_SVM:
+  case (xrtBufferFlags)XCL_BO_FLAGS_DEV_ONLY:
+  case (xrtBufferFlags)XCL_BO_FLAGS_HOST_ONLY:
+  case (xrtBufferFlags)XCL_BO_FLAGS_P2P:
+  case (xrtBufferFlags)XCL_BO_FLAGS_EXECBUF:
```

Comment:
Isn't the compile failure is because xrtBufferFlags is of wrong type?  I don't think it should be uint64_t as it is in xrt_bo.h, but it should be unsigned int to match shim layer flags usage.   Let's revert this particular change and fix the typedef.  

Which OS are you using where you have access to gcc10?

Reply:
I am using Ubuntu 20.x. I believe the error happens because `XCL_BO_FLAGS_EXECBUF` is treated as an `int`, 
but `1<<31` would be treated as -1, then cast to uint64_t would equal `0xFFFFFFFFFFFFFFF`.